### PR TITLE
Turkish Language & Cleaning up the language code

### PR DESCRIPTION
**Pull request by Visual917**
### Combined Locales
Created a locales object to link each language code with its data.

### Unified Locale Function
Made a getLocale function. It combines the base language data with any extra bits you add. This means you don't have to write the same code for each language.

### Streamlined Language Functions
en, es, ja, tr etc now just use getLocale. It's cleaner and makes adding more languages easy. (Hopefully.) I would make them JSON files instead of separate ts files though. But I didn't want to overstep any boundaries.

## Added Turkish Language
Created by @selcukemiravci and @Visual917

### CODE REVIEW

**Comment by joaquim-verges** [comment]
thank you @Visual917 ! super helpful - we'll review this and get it in asap :)

File: packages/react-native/src/evm/i18n/strings.ts [hunk 1]
```diff
@@ -2,41 +2,31 @@ import { DeepPartial, deepMerge } from "../types/deepPartial";
 import { _en } from "./en";
 import { _es } from "./es";
 import { _ja } from "./ja";
+import { _tr } from "./tr";
 import { Locale, LocaleType } from "./types";
 
+const locales = { en: _en, es: _es, ja: _ja };
```
Comment (iketw) [inline packages/react-native/src/evm/i18n/strings.ts:8]:
```suggestion
const locales = { en: _en, es: _es, ja: _ja, tr: _tr };
```

**Review by iketw — COMMENTED** [review]
Thanks for working on this!  A minor comment ;)

**Comment by iketw** [comment]
@Visual917 made the changes here: https://github.com/thirdweb-dev/js/pull/1962. Let me know if it looks good to you and I'll merge it :). Thanks for working on this!

**Comment by jnsdls** [comment]
superseded by #1962

**Pull request closed without merging**